pubtools/pulplib/_impl/model/convert.py

```diff
@@ -81,3 +82,16 @@ def frozenlist_or_none_converter(obj, map_fn=(lambda x: x)):
 frozenlist_or_none_sorted_converter = functools.partial(
     frozenlist_or_none_converter, map_fn=lambda x: sorted(set(x))
 )
+
+
+def frozendict_or_none_converter(obj):
+    # Convert object and values to immutable structures.
+    # If the container is copied, it'll try to convert the object again, but that causes an error.
```

Comment:
I've reworded it to hopefully be more clear. When attr.evolve was called, it would rerun the conversion function, which would fail since the value was already a frozendict. It felt a bit weird to have a check for something already being a frozendict in a function that converts dict->frozendict, so I wanted to have a comment to clarify.